appium/options/android/common/adb/adb_options.py

```diff
@@ -0,0 +1,102 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import timedelta
+from typing import Any
+
+from appium.options.common.supports_capabilities import SupportsCapabilities
+
+
+class AdbOptionsDescriptor:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def __get__(self, obj: Any, cls: Any) -> Any:
+        if self.name == "ADB_EXEC_TIMEOUT":
+            value = getattr(obj, "get_capabilities")(self.name)
+            return None if value is None else timedelta(milliseconds=value)
+        return getattr(obj, "get_capabilities")(self.name)
+
+    def __set__(self, obj: Any, value: Any) -> Any:
+        if self.name == "ADB_EXEC_TIMEOUT":
+            getattr(obj, "set_capabilities")(self.name, int(value.total_seconds() * 1000) if isinstance(value, timedelta) else value)    
+        getattr(obj, "set_capabilities")(self.name, value)
+
+
+class AdbExecTimeoutOption(SupportsCapabilities):
+    ADB_EXEC_TIMEOUT = 'adbExecTimeout'
```

Comment:
I assume this property is also redundant. We could simply use `self.__class__.__name__` itself without the `Option` suffix